Deal with an empty event filter string

bobtfish opened the pull request:
If you set an event filter and then blank it later, the API can return an empty event filter string, which is equivalent to no event filter.

RJPearson94 commented:
Hey @bobtfish, thanks for opening this PR its greatly appreciated and thanks for using the provider 🙌 

I was just looking to see if I could add some acceptance tests to replicate the scenario you mentioned and I found a few slight issues with that resource.

I'm going to look at addressing those issues and pulling in your change, so it might take slightly longer before all the changes are in the main branch and a new release is published to the terraform registry.

bobtfish commented:
> Hey @bobtfish, thanks for opening this PR its greatly appreciated and thanks for using the provider 🙌

No problem. I'd started writing my own when yours popped up and it's generally nicer and more fully featured than mine is, so expect ongoing patches etc.

> I was just looking to see if I could add some acceptance tests to replicate the scenario you mentioned and I found a few slight issues with that resource.
> 
> I'm going to look at addressing those issues and pulling in your change, so it might take slightly longer before all the changes are in the main branch and a new release is published to the terraform registry.

Sounds good, no great rush. Thanks!

RJPearson94 commented:
hey @bobtfish,
 
Apologies for the delay in getting this released. The work to address the issues found has snowballed, and I have decided to clean up some legacy issues at the same time. Hence the delay in this being released.

I only have a few more services left and I'm currently aiming to have a release out next week (w.c. 12th April)

Apologies again and thank you for your patience.

Kind Regards
Rob

RJPearson94 commented:
Hey @bobtfish,

Your change has been released as part of v0.9.0 🎉 . This release has been pushed to the Terraform registry.

Thanks for the fix, it is greatly appreciated 👍 

Apologies again for the delay in getting this release out.

Kind Regards
Rob
